test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js

```diff
@@ -1085,6 +1085,34 @@ describe('test/unit/lib/plugins/aws/deploy/lib/checkForChanges.test.js', () => {
     expect(serverless.service.provider.shouldNotDeploy).to.equal(true);
   });
 
+  it('should print a warning if missing lambda:GetFunction permission', async () => {
+    const { stdoutData } = await runServerless({
+      fixture: 'checkForChanges',
+      cliArgs: ['deploy'],
+      env: { AWS_CONTAINER_CREDENTIALS_FULL_URI: 'ignore' },
+      lastLifecycleHookName: 'aws:deploy:deploy:checkForChanges',
+      awsRequestStubMap: {
+        ...commonAwsSdkMock,
+        Lambda: {
+          getFunction: (() => {
+            const headObjectStub = sandbox.stub();
+            headObjectStub.throws({ providerError: { statusCode: 403 } });
+            return headObjectStub;
+          })(),
```

Comment:
I think you can reduce  this  to :
```suggestion
          getFunction: sandbox.stub().throws({ providerError: { statusCode: 403 } }),
```